src/test/java/com/aws/greengrass/disk/spool/DiskSpoolDAOTest.java

```diff
@@ -78,21 +83,23 @@ void GIVEN_request_with_text_WHEN_operation_to_spool_fail_and_DB_corrupt_THEN_sh
 
     @Test
     void GIVEN_request_with_text_WHEN_operation_to_spool_fail_and_transient_error_THEN_should_retry(ExtensionContext context)
-            throws SQLException, IOException {
+            throws SQLException, IOException, NoSuchFieldException, IllegalAccessException {
         ignoreExceptionOfType(context, SQLTransientException.class);
         SQLException sqlException = new SQLTransientException("Some Transient Error");
-        lenient().when(paths.workPath(anyString())).thenReturn(currDir);
-        lenient().when(connection.prepareStatement(anyString())).thenReturn(preparedStatement);
-        lenient().when(connection.createStatement()).thenReturn(statement);
+        lenient().when(mockPaths.workPath(anyString())).thenReturn(tempDir);
+        lenient().when(dbConnection.prepareStatement(anyString())).thenReturn(preparedStatement);
+        lenient().when(dbConnection.createStatement()).thenReturn(statement);
         // Fail the first two times to check retry behavior
         lenient().when(preparedStatement.executeUpdate())
                 .thenThrow(sqlException)
                 .thenThrow(sqlException)
                 .thenReturn(0);
         lenient().when(statement.executeUpdate(anyString())).thenReturn(0);
 
-        DiskSpoolDAO diskSpoolDAO = spy(new DiskSpoolDAO(paths));
-        doReturn(connection).when(diskSpoolDAO).getDbInstance();
+        DiskSpoolDAO diskSpoolDAO = spy(new DiskSpoolDAO(mockPaths));
+        Field field = DiskSpoolDAO.class.getDeclaredField("dbConnection");
+        field.setAccessible(true);
+        field.set(diskSpoolDAO, dbConnection);
```

Comment:
as mentioned offline, this overwrites the field but doesn't close the existing connection.  while we should be using a real connection in these tests, continuing with mocks will be simpler for now and we can follow up later.

we should add a new constructor in DiskSpoolerDAO that takes a `Supplier<Connection>` as a param.  And DiskSpoolerDAO#dbConnection field can be changed to `Supplier<Connection>`.  this way, when connections are remade during corruption, etc, they will always use the mock connection